lib/lotus/commands/generate/abstract.rb

```diff
@@ -0,0 +1,75 @@
+require 'lotus/environment'
+require 'thor'
+
+module Lotus
+  module Commands
+    class Generate < Thor
+      class Abstract
+        VALID_TEST_TYPES = ['minitest', 'rspec'].freeze
+
+        attr_reader :options, :base_path
+
+        def initialize(options)
+          @options = Lotus::Utils::Hash.new(options).symbolize!
+          assert_options!
+          # TODO: find a good name for this...
+          @base_path = Pathname.pwd
+        end
+
+        def start
+          raise NotImplementedError
+        end
+
+        private
+
+        def lotusrc_options
+          @lotusrc_options ||= Lotusrc.new(Pathname.new(base_path)).read
+        end
+
+        def environment
+          @environment ||= Lotus::Environment.new(options)
+        end
+
+        def template_source_path
+          # TODO: how to do this clean...
+          generator = self.class.name.split('::').last.downcase
+          Pathname.new(::File.dirname(__FILE__) + "/../../generators/#{generator}/").realpath
+        end
+
+        # @since x.x.x
+        # @api private
+        def test_type
+          options.fetch(:test, default_test_type)
+        end
+
+        def default_test_type
+          lotusrc_options.fetch(:test)
+        end
+
+        def template_engine
+          options.fetch(:template, default_template_engine)
+        end
+
+        def default_template_engine
+          lotusrc_options.fetch(:template)
+        end
+
+        # @since x.x.x
+        # @api private
+        def assert_test_type!
```

Comment:
I would choose a better naming for those `assert_*` methods, don't know you but I always think it's a test :)

What about `ensure`? `ensure_valid_test_framework` would very clear!

Also, it's better to avoid `!` when there is no no-bang method: https://github.com/bbatsov/ruby-style-guide#dangerous-method-bang